roofit/roofitcore/src/FitHelpers.cxx

```diff
@@ -66,6 +67,21 @@ constexpr int extendedFitDefault = 2;
 /// \param[in] data The dataset that was used for the fit.
 int calcAsymptoticCorrectedCovariance(RooAbsReal &pdf, RooMinimizer &minimizer, RooAbsData const &data)
 {
+   RooFormulaVar logpdf("logpdf", "log(pdf)", "log(@0)", pdf);
+   RooArgSet obs;
+   logpdf.getObservables(data.get(), obs);
+
+   // Warning if the dataset is binned. TODO: in some cases,
+   // people also use RooDataSet to encode binned data,
+   // e.g. for simultaneous fits. It would be useful to detect
+   // this in this future as well.
+   if (dynamic_cast<RooDataHist const *>(&data)) {
+      oocoutW(&pdf, InputArguments)
+         << "RooAbsPdf::fitTo(" << pdf.GetName()
+         << ") WARNING: Asymptotic error correction is requested for a binned data set. "
```

Comment:
Hi @langenbruch 
Thank you for the PR. I don't agree with this statement. A chi2 fit can give a more correct parameter estimate, but it can also give more biased results, especially in bins with low statistics. We could instead warn that other methods for estimating uncertainty (bootstrapping using toys) are preferable. 
Also, can you please elaborate more why you think the asymptotic procedure does not work for a binned data set. A binned data set is statistically equivalent to a weighted unbinned data set. It is not more a question on the number of bins to be large enough in order to be in the asymptotic regime?

Reply:
Dear @lmoneta , first of all let me say that I am fine with removing the second part of the sentence about the chi2 fit, as my main point is that I do not want people to use the code for a binned fit. I do not know in detail how roofit handles binned fits internally, and do not want to assume responsibility for people using the code for things it was not intended to do.

On your comment about chi2 fits being potentially biased at low statistics I have to say that I do not disagree in general, but it is slightly besides the point. The asymptotically correct method relies on asymptotic statistics, ie. it is valid in the high statistics limit. In this limit I believe you will indeed get the usual chi2 expression when evaluating the expectation values. The asymptotically correct method is not a method of coverage correction at low statistics, for this a proper Neyman construction would be required (both in the weighted and the unweighted case). The point is that the previous Sumw2 method does not provide asymptotically correct intervals for unbinned fits, even in the high statistics limit (as you can easily see when you run the example I implemented).